### DISCUSSION TITLE
DFRobot Router board with other Distros

### DISCUSSION THREAD

**Discussion by CalumDay** [geerlingguy/raspberry-pi-pcie-devices#135] [Q&A] [answered]
Has anyone had any luck using anything other than the OpenWRT image provided? The NIC is a Realtek R8111, the drivers for which (R8169) dont seem to be available on the RPi kernel in Ubuntu LTS, RPi OS or Alpine.

Any suggestions short of rebuilding the Kernel?

**Comment by geerlingguy** [comment, marked as answer]
@Fetish-Voltron - If it's using a Realtek chip, most of them are not covered in Pi OS's default packaged modules, so you'll likely need to recompile the kernel to use it. I think OpenWRT intentionally compiles in pretty much all network drivers, so it's a bit easier there :)

Also linking https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/114 — I'm going to be testing the device this week.

**Reply by CalumDay** [reply]
I figured as much, just hoped I'd missed something obvious. Makes it a peculiar choice for a NIC on a carrier board IMO, especially one advertised as supporting other Distros.

I look forward to seeing your thoughts on it though.

**Reply by CalumDay** [reply]
r8168-dkms is now functional in Ubuntu server 21.04. Previously it failed to build.

**Comment by aduewel** [comment]
@Fetish-Voltron see https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/114#issuecomment-861864051
